File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.py
from itertools import product

def solution(word):
    arr = ["A", "E", "I", "O", "U"]
    words = []
    ans = 0
    
    for i in range(1, 6):
        for j in product(arr, repeat=i):
            words.append("".join(list(j)))

    words.sort()
    
    return words.index(word) + 1

File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.js
function solution(word) {
    let ans = 0;
    let flg = false; // DFS 중지 플래그
    const alpha = ["A", "E", "I", "O", "U"]; // 사전 배열
    
    const DFS = (str) => {
        if(flg || str.length > 5) return; // 중지 플래그가 true거나 문자열 길이가 5 이상이라면 return
        
        if(word === str) {
            // 사전의 단어와 동일하다면
            flg = true; // DFS 중지
            return;
        }
        ans += 1; // 사전 순서 + 1
        for(let i = 0; i < alpha.length; i++){
            // alpha배열 순회하면서 재귀 
            DFS(str + alpha[i]);
        }
    }
    
    DFS("")
    
    return ans;
}